homeassistant/components/light/lightwave.py

```diff
@@ -0,0 +1,112 @@
+"""
+homeassistant.components.light.lightwave.
+
+Implements LightwaveRF lights.
+
+
+My understanding of the LightWave Hub is that devices cannot be discovered
+so must be registered manually. This is done in the configuration file:
+
+switch:
+  - platform: lightwave
+    devices:
+      R1D2:
+        name: Room one Device two
+      R2D1:
+        name: Room two Device one
+
+Each device requires an id and a name. THe id takes the from R#D#
+where R# is the room number and D# is the device number.
+
+For more details on the api see: https://api.lightwaverf.com/
+"""
+import logging
+import voluptuous as vol
+from homeassistant.const import CONF_DEVICES, CONF_NAME
+from homeassistant.components.light import (
+    Light, ATTR_BRIGHTNESS, SUPPORT_BRIGHTNESS, PLATFORM_SCHEMA)
+import homeassistant.helpers.config_validation as cv
+
+DEVICE_SCHEMA = vol.Schema({
+    vol.Required(CONF_NAME): cv.string
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_DEVICES, default={}): {cv.string: DEVICE_SCHEMA}
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+LIGHTWAVE_LINK = 'lightwave_link'
+DEPENDENCIES = ['lightwave']
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Find and return LightWave lights."""
+    lights = []
+    lwlink = hass.data[LIGHTWAVE_LINK]
+
+    for device_id, device_config in config.get(CONF_DEVICES, {}).items():
+        name = device_config[CONF_NAME]
+        lights.append(LRFLight(name, device_id, lwlink))
+
+    async_add_entities(lights)
+
+
+class LRFLight(Light):
+    """Provides a LightWave light."""
+
+    def __init__(self, name, device_id, lwlink):
+        """Initilize the component with the name, device_id and link."""
```

Comment:
It's a platform. Also, docstring should looks like a signature.

Reply:
Done